src/components/Menu/MenuItem.tsx

```diff
@@ -153,18 +220,69 @@ class MenuItem extends UIComponent<Extendable<MenuItemProps>, MenuItemState> {
           ...accessibility.keyHandlers.root,
         },
         overrideProps: () => ({
-          children: menuItemInner,
+          children: (
+            <>
+              {menuItemInner}
+              {maybeSubmenu}
+            </>
+          ),
+          onClick: this.handleClick,
+          onBlur: this.handleWrapperBlur,
         }),
       })
     }
     return menuItemInner
   }
 
+  private handleWrapperBlur = e => {
+    if (!this.props.inSubmenu && !e.currentTarget.contains(e.relatedTarget)) {
+      this.setState({ menuOpen: false })
+    }
+  }
+
   protected actionHandlers: AccessibilityActionHandlers = {
     performClick: event => this.handleClick(event),
+    openMenu: event => this.openMenu(event),
+    closeAllMenus: event => this.closeAllMenus(event),
+    closeMenuAndFocusNextParentItem: event => this.closeMenuAndFocusNextParentItem(event),
+    closeMenu: event => this.closeMenu(event),
+  }
+
+  private updateOutsideClickSubscription() {
+    this.outsideClickSubscription.unsubscribe()
+
+    if (this.props.menu && this.state.menuOpen) {
+      setTimeout(() => {
+        this.outsideClickSubscription = EventStack.subscribe('click', this.outsideClickHandler)
+      })
+    }
+  }
+
+  private outsideClickHandler = e => {
+    if (
+      !doesNodeContainClick(this.itemRef.current, e) &&
+      !doesNodeContainClick(this.menuRef.current, e)
+    ) {
+      this.state.menuOpen && this.trySetState({ menuOpen: false })
```

Comment:
nit: we can make this check (`this.state.menuOpen`) as the first step, because (`doesNodeContainClick`) are less efficient ones:

```js
if (!this.state.menuOpen) {
  return 
}
...
```